crates/hyperqueue/src/server/autoalloc/process.rs

```diff
@@ -1,255 +1,646 @@
-use futures::future::join_all;
 use std::future::Future;
-use std::time::SystemTime;
+use std::ops::ControlFlow;
+use std::path::PathBuf;
+use std::time::{Duration, SystemTime};
+
+use futures::future::join_all;
+use tempdir::TempDir;
+
+use tako::common::{Map, Set};
+
+use tako::messages::gateway::LostWorkerReason;
+use tako::WorkerId;
+
+use crate::common::manager::info::{ManagerInfo, ManagerType};
+use crate::common::rpc::RpcReceiver;
 
-use crate::server::autoalloc::descriptor::SubmitMode;
+use crate::get_or_return;
+use crate::server::autoalloc::queue::pbs::PbsHandler;
+use crate::server::autoalloc::queue::slurm::SlurmHandler;
+use crate::server::autoalloc::queue::{AllocationExternalStatus, QueueHandler, SubmitMode};
+use crate::server::autoalloc::service::AutoAllocMessage;
 use crate::server::autoalloc::state::{
-    Allocation, AllocationEvent, AllocationStatus, DescriptorState, RateLimiterStatus,
+    AllocationState, AutoAllocState, QueueState, RateLimiter, RateLimiterStatus,
 };
-use crate::server::autoalloc::{AllocationId, AutoAllocResult, DescriptorId, QueueInfo};
-use crate::server::event::storage::EventStorage;
+use crate::server::autoalloc::{Allocation, AllocationId, AutoAllocResult, QueueId, QueueInfo};
 use crate::server::job::Job;
 use crate::server::state::{State, StateRef};
-use crate::transfer::messages::{JobDescription, TaskDescription};
+use crate::transfer::messages::{
+    AllocationQueueParams, JobDescription, QueueData, TaskDescription,
+};
+
+pub async fn autoalloc_process(
+    state_ref: StateRef,
+    mut autoalloc: AutoAllocState,
+    mut receiver: RpcReceiver<AutoAllocMessage>,
+) {
+    let max_wait_for_message = Duration::from_secs(60 * 60);
+    loop {
+        match tokio::time::timeout(max_wait_for_message, receiver.recv()).await {
```

Comment:
When there will be a steady stream of message then refresh is not called and old directory will not be deleted (remove_inactive_directories)?

Reply:
Refresh is called after each received message. I made this more explicit.